python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -1031,55 +1017,23 @@ def _getIOTimeOut(self):
             _timeOut = 5. # 5sec hardcoded default
         return _timeOut
 
-    def _getWriteAccess(self):
-        """ tries to get write access to the object.
-        Raise LockingError (or so) on fail """
-        root = self._getRoot()
-        reg = root._getRegistry()
+    def _getSessionLock(self):
```

Comment:
The old code called `_load` in the `finally` part of the `try` block on the registry lock so it was going to load anyway. I agree that at the moment, if you load an object then you acquire the session lock and don't release it until shutdown, however, that would be an even bigger change and should probably wait for another PR!

This PR uses the `@complete_lock` on load where previously it was the `@syncronised' lock.